### PR TITLE
improve tests with assert and a test for flatbuffers as well as add HEVC support for both recording and the CLI extractor

### PR DESCRIPTION
**Pull request by abryden**
Starting to look at H264/H265 support. Considering extending the capture application to support the choice of protobuf and flatbuffers. As part of this extended the tests to make sure flatbuffers are working as intended.

### CODE REVIEW

**Comment by abryden** [comment]
Converting to a draft since I'm attempting to get Initial HEVC support in this PR.

**Comment by abryden** [comment]
As of now this uses Vikas' code to encode H265/HEVC but only serializes the actual H265 frame and not the header/attachments. I'll add a discussion of this over in the discussions section but will likely go ahead and add a header section to the protobuf and flatbuffers definitions barring any objections.

**Comment by jrpowers** [comment]
Thanks Aaron! One small issue I spot is that there are still some "Copyright © 2020 Vikas Reddy. All rights reserved" in the LT files (LT is the abbreviation of his company name LightTwist).

We should definitely make sure to credit Vikas but I think we should remove that copyright line and probably rename the files away from LTXXX just to avoid confusion.

File: common/LTVideoEncodingAndDecoding.swift [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+                //kCVPixelBufferOpenGLESCompatibilityKey: true as CFTypeRef, //TODO LOOK INTO,
+                kCVPixelBufferMetalCompatibilityKey: true as CFTypeRef, //TODO LOOK INTO,
+                //kCVPixelBufferOpenGLESTextureCacheCompatibilityKey: true as CFTypeRef, //TODO LOOK INTO,
+            ]
+            
+            let decodingProperties = [
+                kVTDecompressionPropertyKey_RealTime: true as CFTypeRef,
+                
+                //the following only seem to work on OSX? Any way to force hardware acceleration?
+                //kVTVideoDecoderSpecification_EnableHardwareAcceleratedVideoDecoder
```
Comment (vikasreddy-lighttwist) [inline common/LTVideoEncodingAndDecoding.swift:418]: Not urgent for now, but might want to enable HW acceleration if on OSX here (on iOS I guess it always just uses HW acceleration).

**Comment by vikasreddy-lighttwist** [comment]
Looks good to me, I could see having a setting to force HW encode and decode on OSX (see comment inline on decode).

**Comment by abryden** [comment]
> Looks good to me, I could see having a setting to force HW encode and decode on OSX (see comment inline on decode).

Thanks!  Got the copyright and naming changes in. Working on testing HW decode and encode. I'll report back once I have some evidence on numbers.

File: common/LTVideoEncodingAndDecoding.swift (hunk 1 above)
Comment (abryden) [reply common/LTVideoEncodingAndDecoding.swift]: Good catch! I enabled HW acceleration for encode and decode on macOS . Some non-rigorous testing seemed to show an improvement to CPU utilization in decode. I haven't tested capture on macOS yet other than as an iPad app on the M1 (amusingly uses my logitech webcam rather poorly from within the app on M1 and crashes!)

**Comment by vikasreddy-lighttwist** [comment]
Hey Aaron, would you mind changing the note at the top of those files to:

// Thanks to Vikas Reddy/LightTwist who generously allowed us to use this code 
 // Taken from MIT licensed code created by Vikas Reddy/LightTwist  
 //
 //  Created by Vikas Reddy on 6/22/20.
 //
 
 Might be worthwhile to note the MIT license. Later I can publish that code via a LightTwist repo and you could even link to it maybe just so it's clear / good record keeping.

**Comment by abryden** [comment]
> Hey Aaron, would you mind changing the note at the top of those files to:
> 
> // Thanks to Vikas Reddy/LightTwist who generously allowed us to use this code
> // Taken from MIT licensed code created by Vikas Reddy/LightTwist
> //
> // Created by Vikas Reddy on 6/22/20.
> //
> 
> Might be worthwhile to note the MIT license. Later I can publish that code via a LightTwist repo and you could even link to it maybe just so it's clear / good record keeping.

Done. See commit at https://github.com/arcturus-industries/arcfile/pull/8/commits/7225206a261ed38284cd2bf546c74f27a903b0cd

**Pull request merged**